test/unittests/t_catalog_mgr_rw.cc

```diff
@@ -118,4 +131,98 @@ TEST_F(T_CatalogMgrRw, CloneTree) {
   EXPECT_TRUE(dirent.IsNestedCatalogRoot());
 }
 
+
+TEST_F(T_CatalogMgrRw, SwapNestedCatalog) {
+  CatalogTestTool tester("swap_nested_catalog");
+  EXPECT_TRUE(tester.Init());
+
+  DirSpec spec = MakeBaseSpec();
+  EXPECT_TRUE(tester.ApplyAtRootHash(tester.manifest()->catalog_hash(), spec));
+
+  catalog::WritableCatalogManager *catalog_mgr = tester.catalog_mgr();
+
+  // Look up sub1 and sub2 nested catalogs
+  PathString path;
+  shash::Any sub1_hash;
+  shash::Any sub2_hash;
+  uint64_t sub1_size;
+  uint64_t sub2_size;
+  EXPECT_TRUE(catalog_mgr->LookupNested(PathString("/dir/dir/dir/sub1"),
+                                        &path, &sub1_hash, &sub1_size));
+  EXPECT_TRUE(catalog_mgr->LookupNested(PathString("/dir/dir/dir/sub2"),
+                                        &path, &sub2_hash, &sub2_size));
+
+  // Swap sub1 with itself
+  DirectoryEntry dirent;
+  catalog_mgr->DetachNested();
+  catalog_mgr->SwapNestedCatalog("dir/dir/dir/sub1", sub1_hash, sub1_size);
+  EXPECT_TRUE(catalog_mgr->LookupPath("/dir/dir/dir/sub1/file1",
+                                      kLookupSole, &dirent));
+  EXPECT_STREQ(g_hashes[6], dirent.checksum().ToString().c_str());
+
+  // Swap sub1 and sub2
+  catalog_mgr->DetachNested();
+  catalog_mgr->SwapNestedCatalog("dir/dir/dir/sub1", sub2_hash, sub2_size);
+  catalog_mgr->SwapNestedCatalog("dir/dir/dir/sub2", sub1_hash, sub1_size);
+  EXPECT_TRUE(catalog_mgr->LookupPath("/dir/dir/dir/sub1/file2",
+                                      kLookupSole, &dirent));
+  EXPECT_STREQ(g_hashes[7], dirent.checksum().ToString().c_str());
+  EXPECT_TRUE(catalog_mgr->LookupPath("/dir/dir/dir/sub2/file1",
+                                      kLookupSole, &dirent));
+  EXPECT_STREQ(g_hashes[6], dirent.checksum().ToString().c_str());
+}
+
+
+TEST_F(T_CatalogMgrRw, SwapNestedCatalogFailSlow) {
```

Comment:
This test should not take longer than 1-2 seconds, does it? The `EXPECT_DEATH` are expensive but there are no network timeouts as far as I can see.  Let's make it a regular test then so that it runs routinely.

```suggestion
TEST_F(T_CatalogMgrRw, SwapNestedCatalogFail) {
```

Reply:
It seems to take about 10 seconds.  I based the choice upon the existing `CloneTreeFailSlow` test within the same file, which AFAICT is "slow" only in the sense of using `EXPECT_DEATH()`.

I'll make the change - let me know if it should be "slow" after all.

Reply:
Oh, it takes _that_ long.  Apologies, then the `Slow` suffix is of course appropriate.